hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieLogRecordReader.java

```diff
@@ -410,6 +418,34 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
     }
   }
 
+  private static void prepareRecordPositions(String blockPositionStr,
+                                             List<Integer> blockPositions,
+                                             Set<Integer> globalPositions,
+                                             boolean positionBasedMergeEnabled,
+                                             boolean useEventOrdering) throws IOException {
+    if (!positionBasedMergeEnabled || blockPositionStr == null || blockPositionStr.isEmpty()) {
+      return;
+    }
+
+    Roaring64NavigableMap posBitmap = PositionUtils.deserializeCodecStringWithRoaring64NavigableMap(blockPositionStr);
+    Iterator<Long> iterator = posBitmap.iterator();
+    while (iterator.hasNext()) {
+      long pos = iterator.next();
```

Comment:
if the positions are longs, should we be storing logs in the `blockPositions` and `globalPositions`? Are we worried about casting and boxing overhead here?

Reply:
@the-other-tim-brown , probably not. Max value of integer is more than 2 billion, which is the number of the rows in a base file. Given the default setting for a base file size (128MB), it is impossible to have row number > 2 billion since 2 billion rows * row_size (> 1Byte) > 2 ^ 30 = 2GB (without meta columns). But not sure if a user wants to have a really huge base file or not.

Reply:
I think we should anticipate that the user could have large files

Reply:
2 billion is probably larger than we will see. Are we worried about casting and then boxing the int to integer for each row in the file from a performance point of view?